content/post/gld.md

```diff
@@ -0,0 +1,1363 @@
+---
+author: vadim
+date: 2018-04-17
+title: "go-license-detector"
+image: /post/gld/wtfpl.png
+description: "Detecting the license of an open source projects is harder than it seems. We have created go-license-detector, a Go library and command line application to solve that task."
+categories: ["science", "technical"]
+---
+
+While working on [Public Git Archive](https://github.com/src-d/datasets/PublicGitArchive), we
+thought that it would be handy to include the license of each project in the index file, so that
+people could easily filter "grey" repositories without a clear license. Besides, we were
+curious about the licenses distribution. GitHub already detects licenses by leveraging
+[benbalter/licensee](https://github.com/benbalter/licensee) Ruby library and the easy solution was
+to query GitHub API. However, we were not satisfied with
+its detection quality: many projects which actually contain the license file in a non-standard
+format are missed, and some are misclassified. This is how
+[go-license-detector](https://github.com/src-d/go-license-detector) was born.
+
+The goals were defined from the very beginning:
+
+1. Favor false positives over false negatives (target data mining instead of compliance).
+2. Perform fast.
+3. Detect as many licenses as possible on the [hand-collected dataset of 1,000 top-starred repositories
+on GitHub](https://github.com/src-d/go-license-detector/blob/master/licensedb/dataset.zip).
+4. Comply with SPDX [licenses list](https://github.com/spdx/license-list-data) and
+[detection guidelines](https://spdx.org/spdx-license-list/matching-guidelines).
+
+(1) means that we should rather label a project with a bit inaccurate license than miss its
+license completely. The open source compliance departments will not be satisfied with this choice,
+as they need the opposite: the missed projects are manually studied. (2) restricts from using a
+scripting language such as Python or Ruby, and we chose Go for our implementation. (3) leads
+to technical tricks, hacks and heuristics which result in complex code.
+(4) is the only way to obtain the database of 400 different licenses validated by professional
+lawyers.
+
+The following table compares the current go-license-detector with GitHub's built-in license detector,
+Google's licenseclassifier and Ben Boyter's `lc` on the
+[reference 1k dataset](https://github.com/src-d/go-license-detector/blob/master/licensedb/dataset.zip):
+
+|Detector|Detection rate|Time to scan, sec|
```

Comment:
I would consider creating a 2D diagram of these metrics, where you compare accuracy and speed.
This would show how much better we perform while being faster.

Reply:
Once I have time I add this